rerun_py/rerun_sdk/rerun/_image.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+import io
+import pathlib
+from enum import Enum
+from typing import IO, Iterable
+
+import numpy as np
+from PIL import Image as PILImage
+
+from ._log import AsComponents, ComponentBatchLike
+from .archetypes import Image
+from .components import DrawOrderArrayLike, TensorData
+from .datatypes import TensorBuffer, TensorDimension
+
+
+class ImageFormat(Enum):
+    """Image file format."""
+
+    BMP = "BMP"
+    """BMP format."""
+
+    GIF = "GIF"
+    """GIF format."""
+
+    JPEG = "JPEG"
+    """JPEG format."""
+
+    PNG = "PNG"
+    """PNG format."""
+
+    TIFF = "TIFF"
+    """TIFF format."""
+
+    def __str__(self) -> str:
+        return self.name
+
+
+class ImageEncoded(AsComponents):
+    def __init__(
+        self,
+        *,
+        path: str | pathlib.Path | None = None,
+        contents: bytes | IO[bytes] | None = None,
+        format: ImageFormat | None = None,
+        draw_order: DrawOrderArrayLike | None = None,
+    ) -> None:
+        """
+        A monochrome or color image encoded with a common format (PNG, JPEG, etc).
+
+        Parameters
+        ----------
+        path:
+            A path to an image file stored on the local filesystem.
+        contents:
+            The contents of the image file. Can be a BufferedReader, BytesIO, or bytes.
+        format:
+            The format of the image file. If not provided, it will be inferred from the file extension.
+        draw_order:
+            An optional floating point value that specifies the 2D drawing order.
+            Objects with higher values are drawn on top of those with lower values.
+        """
+        if len([x for x in (path, contents) if x is not None]) != 1:
```

Comment:
🤯 
totally fine by me, but why not just `if path is not None and contents is not None` ;)

Reply:
Because that doesn't handle the case of them both being None.
```
if (path is not None and contents is not None) or (path is None and contents is None):
```